code/model/EditableCustomRule.php

```diff
@@ -147,4 +147,100 @@ public function canUnpublish($member = null)
     {
         return $this->canDelete($member);
     }
+
+	/**
+	 *
+	 * @return array
+	 */
+	public function BuildExpression()
+	{
+		$rule = $this;
+		/** @var EditableFormField $formFieldWatch */
+		$formFieldWatch = $this->ConditionField();
+		//Encapsulated the action to the object
+		$action = $formFieldWatch->getJsEventHandler();
+
+		// is this field a special option field
+		$checkboxField = $formFieldWatch->isCheckBoxField();
+		$radioField = $formFieldWatch->isRadioField();
+
+		$target = sprintf('$("%s")',$formFieldWatch->getSelectorFieldOnly());
+
+		// and what should we evaluate
+		switch($rule->ConditionOption)
+		{
+			case 'IsNotBlank':
+				$expression = ($checkboxField || $radioField) ? "{$target}.is(\":checked\")" : "{$target}.val() != ''";
+
+				break;
+			case 'IsBlank':
+				$expression = ($checkboxField || $radioField) ? "!({$target}.is(':checked'))" : "{$target}.val() == ''";
+
+				break;
+			case 'HasValue':
+				if($checkboxField)
+				{
+					$expression = "{$target}.prop('checked')";
+				}elseif($radioField)
+				{
+					// We cannot simply get the value of the radio group, we need to find the checked option first.
+					$expression = sprintf('%s.closest(".field, .control-group").find("input:checked").val()=="%s"',$target,$rule->FieldValue);
+				}else
+				{
+					$expression = sprintf('%s.val() == "%s"',$target,$rule->FieldValue);
+				}
+
+				break;
+			case 'ValueLessThan':
+				$expression = sprintf('%s.val() < parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueLessThanEqual':
+				$expression = sprintf('%s.val() <= parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueGreaterThan':
+				$expression = sprintf('%s.val() > parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueGreaterThanEqual':
+				$expression = sprintf('%s.val() >= parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			default: // ==HasNotValue
+				if($checkboxField)
+				{
+					$expression = sprintf('!%s.prop("checked")',$target);
+				}elseif($radioField)
+				{
+					// We cannot simply get the value of the radio group, we need to find the checked option first.
+					$expression = sprintf('%s.parents(".field, .control-group").find("input:checked").val()!="%s"',$target,$rule->FieldValue );
+				}else
+				{
+					$expression = sprintf('%s.val() != "%s"',$target,$rule->FieldValue);//'$(this).val() != "' . $rule->FieldValue . '"';
+				}
+
+				break;
+		}
+
+		$result = array(
+			'operation'      => $expression,
+			'event'          => $action,
+		);
+		return $result;
```

Comment:
May as well just return this array instead of assigning to a variable, _unless_ you want to provide an extension point here (which might be a good idea)